pkg/apis/pipeline/v1/result_validation.go

```diff
@@ -51,3 +53,25 @@ func (tr TaskResult) Validate(ctx context.Context) (errs *apis.FieldError) {
 
 	return nil
 }
+
+
+func validateObjectResult(tr TaskResult) (errs *apis.FieldError) {
+	if ParamType(tr.Type) == ParamTypeObject && tr.Properties == nil {
+		return apis.ErrMissingField(fmt.Sprintf("%s.properties", tr.Name))
+	}
+
+	invalidKeys := []string{}
+	for key, propertySpec := range tr.Properties {
+		if propertySpec.Type != ParamTypeString {
+			invalidKeys = append(invalidKeys, key)
+		}
+	}
+
+	if len(invalidKeys) != 0 {
+		return &apis.FieldError{
+			Message: fmt.Sprintf("The value type specified for these keys %v is invalid", invalidKeys),
```

Comment:
maybe include in the message that properties must be strings?

Reply:
sure